src/main/kotlin/com/atlassian/performance/tools/jiraactions/webdriver/WebDriverUtils.kt

```diff
@@ -0,0 +1,48 @@
+package com.atlassian.performance.tools.jiraactions.webdriver
+
+import org.openqa.selenium.JavascriptExecutor
+import org.openqa.selenium.NoSuchElementException
+import org.openqa.selenium.WebDriver
+import org.openqa.selenium.WebElement
+import org.openqa.selenium.support.ui.ExpectedCondition
+import java.util.concurrent.atomic.AtomicInteger
+
+class WebDriverUtils {
+    companion object {
+        fun <T> executeScript(driver: WebDriver, js: String): T {
+            driver as JavascriptExecutor
+            return driver.executeScript(js) as T
+        }
+
+        fun sometimesApply(condition: ExpectedCondition<Boolean>): ExpectedCondition<Boolean> {
+            return object : ExpectedCondition<Boolean> {
+                val callCounter = AtomicInteger()
+
+                override fun apply(input: WebDriver?): Boolean {
+                    return callCounter.incrementAndGet() % 5 == 0 && condition.apply(input)!!
+                }
+
+                override fun toString(): String {
+                    return "Sometimes: $condition"
+                }
+            }
+        }
+
+        fun sometimesLocate(condition: ExpectedCondition<WebElement>): ExpectedCondition<WebElement> {
+            return object : ExpectedCondition<WebElement> {
+                val callCounter = AtomicInteger()
+
+                override fun apply(input: WebDriver?) : WebElement {
+                    if (callCounter.incrementAndGet() % 5 != 0) {
+                        throw NoSuchElementException("We haven't actually looked for $condition")
```

Comment:
```suggestion
                        return null
```

Reply:
This is modelled after presenceOfElementLocated, which throws or returns the element.